### PR TITLE
Chore: Trigger deploy preview only when certain directories changed

### PR DESCRIPTION
**Pull request by literat**
* using custom netlify build hook
  * @see https://docs.netlify.com/configure-builds/build-hooks/

### CODE REVIEW

File: bin/ci/deploy-preview.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+function has_code_changes() {
+  git fetch --no-tags --depth=50 origin main
```
Comment (OndraM) [inline bin/ci/deploy-preview.sh:4]: Whats the purpose of this fetch? If we need depth > 1 (which is `actions/checkout@v2` default), we can use `fetch-depth: 50` right in the build step, see https://github.com/actions/checkout/

File: bin/ci/deploy-preview.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    echo "Code changes detected."
+
+    return 1
+  fi
+}
+
+function trigger_build() {
+  echo "Triggering build..."
+
+  curl -X POST -d '{}' https://api.netlify.com/build_hooks/61bca39f3b89823193476ecd?trigger_branch=$1
```
Comment (OndraM) [inline bin/ci/deploy-preview.sh:21]: Don't you want to put the hook id to secrets? 🤔

**Review by OndraM — CHANGES_REQUESTED** [review]
BTW, is the shell script even necessary? Cannot the same be accomplished right in action config? Like:

```yaml
on:
  push:
    paths:
      - 'src/'
      - 'examples/'
```

https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-including-paths

**Comment by literat** [comment]
Closing this because we are deploying under the LMC account from now.

**Pull request closed without merging**